notebooks/Notebook3.md

```diff
@@ -2,33 +2,56 @@
-improve the project? Did the feedback point out or offer something you hadn't considered?
-Did it help you make a design decision? Was it helpful in addressing the most pressing
-issues in your project? How will you incorporate the feedback into your work? Will you
-change something about the design, implementation, or evaluation as a result?
+“Thanks for the description of the issue and the link to the changes, I think it's important especially in these reflections to make sure you describe even things you get stuck on as it shows that effort was put in during that time, which is really important for a project like this where time sinks can be detrimental to your progress.”
+
+
+“Yeah, if you're trying to run a Java app in codespaces, it can be tricky. I see that you have a .devcontainer.json file that should be set up for Java. We can try to debug it together in class on Wednesday, or during office hours!”
+
+
+“I reviewed some of the files in this folder to look over how your player position classes are set up and I better understand why much of the code was previously described as tedious and how a call like the one you provided above functions. One recommendation I would make would be to rename some of the 'set' methods as 'add' methods (i.e. setINTs => addINTs) as it seems these methods just add to the current count. I don't think this really affects the outcome of what users will be interacting with but it is just a minor detail that could clarify what these functions do.”
+
+
+“One note I would have is that your StringGrouping.java file contains if(groupsMatcher.matches()) which works to prevent errors but it seems like nothing happens if a user inputs a string that doesn't match the regex pattern so I would recommend adding an else case which lets users know they've made an error”
+
+
+“The "run" play could also be referred to as a "rush"”
+
+
+I will address the previous comments from last week. Firstly, I hope you don’t mind the rant but most of this week's notebook will consist of the many many errors I ran into. Secondly, I went into office hours and prof Ben was able to provide me with a layout that has the capabilities of running a java app with java fx. This really helped this week and I appreciate the help from him. Thirdly, to address the set vs add. I can see how just looking over the code you may find that these functions are more related to an add feature function. However in my mind they are more of a set/update function. For the time being I will keep them as a set function title but might change them later. Also they do not really affect the user so like you said it is not at the top of my list. Fourthly, This week I added in error handling so the part of the code if(groupsMatcher.matches()) is actually more relevant now. Also when I worked with regex last summer during my internship that is how we used it so its kinda a habit. Lastly, wow I cannot believe I did not think to use rush. I added that in Thanks!!
```

Comment:
Great response to all the feedback from last week's notebook, I like that you provided a justification for each decision!